### ISSUE TITLE
jax.numpy.std bad behaviour with jax.grad evaluated over constant vector ?

### ISSUE THREAD

**Issue by lfaucheux** [google/jax#11716]
Evaluating (e.g. on colab & jax==0.3.14) the gradient of `jax.numpy.std` over constant vectors return `nan`s.

```python
import jax
import jax.numpy as jnp
print(
    jax.grad(jnp.std)(
        jnp.array([1, 1], dtype=jnp.float64)
    )
)
# DeviceArray([nan, nan], dtype=float64)
```

I go with `jax.numpy.var` for the time being, (returns `DeviceArray([0., 0.], dtype=float64)`).

**Comment by jakevdp** [comment]
The standard deviation involves computing the square root, and the gradient of a square root evaluated at zero diverges:
```python
>>> print(jax.grad(jnp.sqrt)(0.0))
inf
```
So I suspect that `NaN` is the correct answer here.

**Comment by lfaucheux** [comment]
Indeed true. It felt like it should have been `[-0.5  0.5]` though.

**Comment by jakevdp** [comment]
I am vaguely recalling that this issue came up at some point in the past (perhaps in the context of the standard deviation of a length-1 array?) but I'm not able to find the discussion at the moment.
